### ISSUE TITLE
How to save mode checkpoint for best metric values during training (+ validation) pipeline

### ISSUE THREAD

**Issue by Harshagarwal19** [OML-Team/open-metric-learning#578]
Hello, 
First of all thankyou for your amazing work. 
I am new to using open-metric-learning package, I am trying to use it for training a vit-dino based feature extractor model (for image matching task). 
I had a question regarding saving the model checkpoints, when training the model with the pytorch lightning integration. 

```python
metric_callback = MetricValCallback(metric=EmbeddingMetrics(extra_keys=["paths",]), log_images=True);

# 1) Logging with Tensorboard
logger = TensorBoardPipelineLogger(".")

# run
pl_model = ExtractorModule(extractor, criterion, optimizer)
trainer = pl.Trainer(max_epochs=3, callbacks=[metric_callback], num_sanity_val_steps=0, logger=logger)
trainer.fit(pl_model, train_dataloaders=train_loader, val_dataloaders=val_loader)
```

Following is the snippet from your provided reference code for training + validation (with pytorch lightning). The above code saves the model checkpoint for the last epoch only. But I was trying to find a way to save the model checkpoint for the best metric values (example: like best values of `precision`). I even suffered through the source code, but I was unable to figure it out. 
According to me it seems that `metric_callback` object needs to modified somehow but I am unable to figure it out. Can you help me regarding this issue, or just point me to documentation for the same if it is available.

**Comment by AlekseySh** [comment]
Hi, @Harshagarwal19 

Thank you for using OML. 

In this case you just need to use buil-in functionality presented in PyTrch Lightning: https://lightning.ai/docs/pytorch/stable/api/lightning.pytorch.callbacks.ModelCheckpoint.html

When we developed our Pipelines (Config API) we used the same class: https://github.com/OML-Team/open-metric-learning/blob/main/oml/lightning/pipelines/parser.py#L97 

Check it out, it should be simple :)

**Comment by AlekseySh** [comment]
fee free to reopen this issue if you have any questions

**Comment by Harshagarwal19** [comment]
Thanks that was helpful!
